src/material-experimental/mdc-chips/chip-action.ts

```diff
@@ -0,0 +1,154 @@
+      // as the Angular binding in order to maintain consistency.
+      if (name === 'tabindex') {
+        this._updateTabindex(parseInt(value));
+      } else {
+        this._elementRef.nativeElement.setAttribute(name, value);
+      }
+    },
+    removeAttribute: (name: string) => {
+      if (name !== 'tabindex') {
+        this._elementRef.nativeElement.removeAttribute(name);
+      }
+    },
+    getElementID: () => this._elementRef.nativeElement.id,
+    emitEvent: <T>(eventName: string, data: T) => {
+      emitCustomEvent<T>(this._elementRef.nativeElement, this._document, eventName, data, true);
+    },
+  };
+
+  _handleClick() {
+    // Usually these events can't happen while the chip is disabled since the browser won't
+    // allow them which is what MDC seems to rely on, however the event can be faked in tests.
+    if (!this.disabled) {
+      this._foundation.handleClick();
+    }
+  }
+
+  _handleKeydown(event: KeyboardEvent) {
+    // Usually these events can't happen while the chip is disabled since the browser won't
+    // allow them which is what MDC seems to rely on, however the event can be faked in tests.
+    if (!this.disabled) {
+      this._foundation.handleKeydown(event);
+    }
+  }
+
+  protected _createFoundation(adapter: MDCChipActionAdapter): MDCChipActionFoundation {
+    return new MDCChipPrimaryActionFoundation(adapter);
+  }
+
+  constructor(
+    public _elementRef: ElementRef,
+    @Inject(DOCUMENT) _document: any,
+    private _changeDetectorRef: ChangeDetectorRef,
+  ) {
+    super();
+    this._foundation = this._createFoundation(this._adapter);
+
+    // Set the action classes here since they're somewhat long
+    // and error prone, and we need them in several places.
+    const classList = _elementRef.nativeElement.classList;
+    const actionType = this._foundation.actionType();
+    classList.add('mdc-evolution-chip__action', 'mat-mdc-chip-action');
+
+    if (actionType === MDCChipActionType.PRIMARY) {
+      classList.add('mdc-evolution-chip__action--primary');
+    } else if (actionType === MDCChipActionType.TRAILING) {
+      classList.add('mdc-evolution-chip__action--trailing');
+    }
+
+    if (_elementRef.nativeElement.nodeName === 'BUTTON') {
+      _elementRef.nativeElement.setAttribute('type', 'button');
+    }
+  }
+
+  ngAfterViewInit() {
+    this._foundation.init();
+    this._foundation.setDisabled(this.disabled);
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['disabled']) {
+      this._foundation.setDisabled(this.disabled);
+    }
+  }
+
+  ngOnDestroy() {
+    this._foundation.destroy();
+  }
+
+  focus() {
+    this._elementRef.nativeElement.focus();
+  }
+
+  _getFoundation() {
+    return this._foundation;
+  }
+
+  _updateTabindex(value: number) {
+    this.tabIndex = value;
+    this._changeDetectorRef.markForCheck();
+  }
+
+  static ngAcceptInputType_disabled: BooleanInput;
+  static ngAcceptInputType_tabIndex: NumberInput;
```

Comment:
We can drop these now, right?

Reply:
There were some internal issues with https://github.com/angular/components/pull/23606 so I decided to keep them here.